src/Column/ActionColumn.php

```diff
@@ -0,0 +1,458 @@
+
+/**
+ * ActionColumn is a column for the {@see GridView} widget that displays buttons for viewing and manipulating the items.
+ */
+final class ActionColumn extends Column
+{
+    private array $buttons = [];
+    private string $primaryKey = 'id';
+    private string $template = '{view}{update}{delete}';
+    private array $urlArguments = [];
+    private Closure|null $urlCreator = null;
+    private bool $urlEnabledArguments = true;
+    private UrlGeneratorInterface|null $urlGenerator = null;
+    private array $urlParamsConfig = [];
+    private string $urlName = '';
+    private array $urlQueryParameters = [];
+    private array $visibleButtons = [];
+
+    /**
+     * Return new instance with the buttons array.
+     *
+     * @param array $value button rendering callbacks. The array keys are the button names (without curly brackets),
+     * and the values are the corresponding button rendering callbacks. The callbacks should use the following
+     * signature:
+     *
+     * ```php
+     * [
+     *     buttons => [
+     *         'action' => function (string $url, $data, int $key) {
+     *             // return the button HTML code
+     *         }
+     *     ],
+     * ]
+     * ```
+     *
+     * where `$url` is the URL that the column creates for the button, `$data` is the data object being rendered
+     * for the current row, and `$key` is the key of the data in the data provider array.
+     *
+     * You can add further conditions to the button, for example only display it, when the data is editable (here
+     * assuming you have a status field that indicates that):
+     *
+     * ```php
+     * [
+     *     buttons = [
+     *         'update' => function (string $url, $data, $key) {
+     *             return $data->status === 'editable' ? Html::a('Update', $url) : '';
+     *         },
+     *     ],
+     * ],
+     * ```
+     *
+     * @return static
+     */
+    public function buttons(array $value): static
+    {
+        $new = clone $this;
+        $new->buttons = $value;
+
+        return $new;
+    }
+
+    /**
+     * Initializes the default button rendering callback for single button.
+     *
+     * @param string $name The button names as it's written in template.
+     * @param string $icon The icon name.
+     * @param array $attributes The HTML attributes in terms of name-value pairs.
+     *
+     * @return static
+     */
+    public function createDefaultButtons(): static
+    {
+        /** @psalm-var array<string,Closure> */
+        $defaultButtons = [
+            'view' => static fn (string $url): string => A::tag()
+                ->addAttributes(
+                    [
+                        'name' => 'view',
+                        'role' => 'button',
+                        'style' => 'text-decoration: none!important;',
+                        'title' => 'View',
+                    ],
+                )
+                ->content(Span::tag()->content('&#128270;')->encode(false))
```

Comment:
We can use unicode directly, I think i.e. 🔎 It's more readable.

Reply:
Then we can remove `->encode(false)`

Reply:
done.